src/Network.cpp

```diff
@@ -404,13 +404,14 @@ Network::Netresult Network::get_scored_moves(
 Network::Netresult Network::get_scored_moves_internal(
     GameState * state, NNPlanes & planes, int rotation) {
     assert(rotation >= 0 && rotation <= 7);
-    constexpr int channels = INPUT_CHANNELS;
-    assert(channels == planes.size());
+    constexpr int input_channels = INPUT_CHANNELS;
+    assert(input_channels == planes.size());
     constexpr int width = 19;
     constexpr int height = 19;
-    constexpr int max_channels = MAX_CHANNELS;
-    std::vector<float> input_data(max_channels * width * height);
-    std::vector<float> output_data(max_channels * width * height);
+
+    auto convolve_channels = int(conv_pol_w.size() / conv_pol_b.size());
+    std::vector<float> input_data(convolve_channels * width * height);
```

Comment:
Is this actually correct? The point of my previous remark was that this is only ever equal to the input channels, and now you're resizing it back to the same size as the outputs, so that would be too big? Thereby throwing a bunch of your speedup away?